functions/_fzf_search_completions.fish

```diff
@@ -0,0 +1,189 @@
+		end
+
+		# If the common prefix includes a / we are completing a file path.
+		# Strip the prefix until the last / completely and later re-add it on the replaced token
+		set -l path_prefix (string match --regex --groups-only -- '^(.*/)[^/]*$' $common_prefix)
+		if test $status = 0
+			set -l path_prefix_length (string length -- $path_prefix)
+			set -l new_start (math 1 + $path_prefix_length)
+			for i in (seq (count $completions))
+				set completions[$i] (string sub -s $new_start -- $completions[$i])
+				set actual_completions[$i] (string sub -s $new_start -- $actual_completions[$i])
+			end
+
+			# We have a path-like prefix and will therefore strip this common prefix from all
+			# completions to un-clutter the menu.
+			set each_prefix (string sub -l $path_prefix_length -- $common_prefix)
+			set common_prefix_length (math $common_prefix_length - $path_prefix_length)
+			set common_prefix (string sub -s $new_start -- $common_prefix)
+		end
+
+		# Detect whether descriptions are present and the length of each completion.
+		set -l has_descriptions false
+		set -l longest_completion $_fzf_search_completions_min_description_offset
+		for i in (seq (count $completions))
+			if string match --quiet -- "*"\t"*" $completions[$i]
+				set has_descriptions true
+
+				# Here we additionally remember the longest completion to align the descriptions in fzf later
+				set longest_completion (math max $longest_completion,(string length -- $actual_completions[$i]))
+				set completions[$i] $actual_completions[$i]\t(set_color -i yellow)$descriptions[$i](set_color normal)
+			end
+		end
+
+		# FIXME Would technically work, but not sure it's worth the effort to match the descriptions
+		# after filtering.
+		# Remove tokens from the completion list that are already present on the current commandline.
+		#set -l all_tokens (commandline -o)
+		#set -l current_token_index (math 1 + (count (commandline -co)))
+		#set --erase all_tokens[$current_token_index]
+		#set -l remaining (comm --zero-terminated -23 (string join0 -- $actual_completions | psub) (string join0 -- $all_tokens | sort | psub) | string split0)
+
+		# TODO pressing / in a completion should add the completion and immediately start a new completion
+		test $has_descriptions = true; and set -l fzf_complete_description_opts \
+			--tabstop=(math 2 + $longest_completion)
+		set -l fzf_output (
+			string join0 -- $completions \
+			| _fzf_wrapper \
+				--read0 \
+				--print0 \
+				--ansi \
+				--multi \
+				--bind=tab:down,btab:up,change:top,ctrl-space:toggle \
+				--tiebreak=begin \
+				--query=$common_prefix \
+				--print-query \
+				$fzf_complete_description_opts \
+				$fzf_complete_opts \
+			| string split0)
+
+		switch $pipestatus[2]
+			case 0
+				# If something was selected, discard the current query
+				set results $fzf_output[2..-1]
+			case 1
+				# User accepted without selecting anything, thus we will
+				# use just the current query
+				set results $fzf_output[1]
```

Comment:
This comment is better described as "Nothing matched the user's query so just use their query".